nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/CleanSurefireExecution.java

```diff
@@ -79,63 +87,67 @@ public void run() throws MojoExecutionException {
         try {
             Logger.getGlobal().log(Level.CONFIG, this.configuration.toString());
             MojoExecutor.executeMojo(this.surefire, MojoExecutor.goal("test"),
-                    this.createListenerConfiguration((Xpp3Dom) this.surefire.getConfiguration()),
+                    this.setReportOutputDirectory((Xpp3Dom) this.surefire.getConfiguration()),
                     MojoExecutor.executionEnvironment(this.mavenProject, this.mavenSession, this.pluginManager));
         } catch (MojoExecutionException mojoException) {
             Logger.getGlobal().log(Level.INFO, "Surefire failed when running tests for " + this.configuration.executionId);
+
+            SurefireReportParser parser = new SurefireReportParser(
+                    Arrays.asList(this.configuration.getExecutionDir().toFile()), Locale.getDefault());
+            try {
+                Set<String> failingTests = new LinkedHashSet<>();
+                for (ReportTestSuite report : parser.parseXMLReportFiles()) {
+                    for (ReportTestCase testCase : report.getTestCases()) {
+                        if (testCase.hasFailure()) {
+                            String testName = testCase.getFullName();
```

Comment:
Is there a way to get the separate test class and test method names from the ReportTestCase?